Feature/get pokemon details

ChristianAsilo opened the pull request:
- Added pokemon ability and abilities model
- Added pokemon move and moves model
- Added pokemon stat and stats model
- Added pokemon details model
- Added pokemonDetails in main
- Added pokemonDetails in pokemon_api 
![image](https://user-images.githubusercontent.com/128013689/226510796-bfe68600-8228-4455-bcc5-384591743270.png)
![image](https://user-images.githubusercontent.com/128013689/226510810-e90e3dbd-1284-4559-881e-6f1208c0b961.png)
![image](https://user-images.githubusercontent.com/128013689/226510822-41d08f0a-4497-47c5-bcdd-bf33c9aaaea7.png)
![image](https://user-images.githubusercontent.com/128013689/226510842-b94a3aae-8ec4-43ee-bfbb-1d0e3a31283a.png)

chrstn06 committed:
feat: get pokemon details
- Added pokemon ability and abilities model
- Added pokemon move and moves model
- Added pokemon stat and stats model
- Added pokemon details model
- Added pokemonDetails in main
- Added pokemonDetails in pokemon_api

chrstn06 committed:
Merge branch 'review' of github.com:ChristianAsilo/pokedex-async-redux into feature/get-pokemon-details

shawn-ffufm commented on lib/api/handler/pokemon_api.dart:
```diff
@@ -22,4 +23,10 @@ class PokemonApi {
     });
   }
+
+  Future<PokemonDetails> getPokemonDetails({required String index}) async {
```
use pokemon name instead it's much easier than using id/index 🙂

shawn-ffufm commented on lib/api/model/pokemon_abilities.dart:
```diff
@@ -0,0 +1,15 @@
+
+@freezed
+class PokemonAbilities with _$PokemonAbilities {
+  factory PokemonAbilities({
```
kindly remove prefix Pokemon - Ability is enough. also don't forget to update the dart file as well 🙂 

apply as well to those related models except the pokemon_details.dart just keep it as is 🙂

shawn-ffufm replied:
kindly rename class to `Ability` instead 🙂

shawn-ffufm commented on lib/api/model/pokemon_ability.dart:
```diff
@@ -0,0 +1,15 @@
+part 'pokemon_ability.g.dart';
+
+@freezed
+class PokemonAbility with _$PokemonAbility {
```
rename to AbilityInfo instead 🙂

shawn-ffufm replied:
see related feedback above 🙂

shawn-ffufm commented on lib/api/model/pokemon_move.dart:
```diff
@@ -0,0 +1,15 @@
+part 'pokemon_move.g.dart';
+
+@freezed
+class PokemonMove with _$PokemonMove {
```
see related feedback above 🙂

shawn-ffufm commented on lib/api/model/pokemon_moves.dart:
```diff
@@ -0,0 +1,15 @@
+part 'pokemon_moves.g.dart';
+
+@freezed
+class PokemonMoves with _$PokemonMoves {
```
see related feedback above - make it singular as well 🙂

shawn-ffufm commented on lib/api/model/pokemon_stat.dart:
```diff
@@ -0,0 +1,15 @@
+part 'pokemon_stat.g.dart';
+
+@freezed
+class PokemonStat with _$PokemonStat {
```
see related feedback above 🙂

shawn-ffufm commented on lib/api/model/pokemon_stats.dart:
```diff
@@ -0,0 +1,16 @@
+
+@freezed
+class PokemonStats with _$PokemonStats {
+  factory PokemonStats({
```
see related feedback above - make it singular 🙂

shawn-ffufm reviewed (changes requested):
@ChristianAsilo see pr feedback 🙂

chrstn06 committed:
feat: get pokemon details
- Modified models
- Removed prefix Pokemon
- Used pokemon name istead of index
- Resolved other pr feedbacks

shawn-ffufm reviewed (approved):
lgtm 💪
